beacon-chain/blockchain/service_test.go

```diff
@@ -533,6 +566,63 @@ func TestReceiveBlock_RemovesPendingDeposits(t *testing.T) {
 	testutil.AssertLogsContain(t, hook, "Executing state transition")
 }
 
+func TestReceiveBlock_CheckStateRoot(t *testing.T) {
+	hook := logTest.NewGlobal()
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+	chainService := setupBeaconChain(t, false, db, false, nil)
+	deposits, privKeys := setupInitialDeposits(t, 100)
+	eth1Data := &pb.Eth1Data{
+		DepositRootHash32: []byte{},
+		BlockHash32:       []byte{},
+	}
+	beaconState, err := state.GenesisBeaconState(deposits, 0, eth1Data)
+	if err != nil {
+		t.Fatalf("Can't generate genesis state: %v", err)
+	}
+	parentHash, genesisBlock := setupGenesisBlock(t, chainService, beaconState)
+	beaconState.Slot++
+	if err := chainService.beaconDB.UpdateChainHead(genesisBlock, beaconState); err != nil {
+		t.Fatal(err)
+	}
+
+	// good block processing check
```

Comment:
to be consistent with the rest, do you mind separating `good block processing check` and `bad block processing check` to 2 tests? We like having one test to test one thing

Reply:
+1